### PR TITLE
Replace table and visualization by dataset and map

### LINKED ISSUE #2748: Update Editor Menu Names to Match New Dashboard
Change editor menu names that refer to "tables" to "datasets" and "visualizations" to "maps"

Edit Menu prior to sharing visualization:
screen shot 2015-03-12 at 3 04 42 pm

Edit Menu after share visualization:
screen shot 2015-03-12 at 3 03 44 pm

Edit Menu after share visualization in Map View:
screen shot 2015-03-12 at 3 14 21 pm

Visualize Menu
screen shot 2015-03-12 at 3 04 50 pm

Share Menu Option:
screen shot 2015-03-12 at 3 03 20 pm

Visualization Metadata
screen shot 2015-03-12 at 3 03 03 pm

Table metadata
screen shot 2015-03-12 at 3 11 38 pm

### PR DESCRIPTION
**Pull request by xavijam**
Basically that, no more visualization or table texts. From now on:

Visualizations -> Map
Table -> Dataset

---

Parts reviewed:

- [x] Editor:
  - [x] Right panel
  - [x] Item metadata
  - [x] Item privacy
  - [x] Dataset creation dialog
  - [x] Item options
  - [x] Map creation dialog 
  - [x] Share map dialog
- [x] Public table/map

---

Fixes #2748

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2758/))

File: lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js [hunk 1]
```diff
@@ -19,29 +19,29 @@ cdb.admin.MergeTablesModel = cdb.core.Model.extend({
...
     next:       _t("Next"),
     back:       _t("Go back"),
     merge: {
       column: {
         title:  _t("Column Join"),
-        desc:   _t("Create a new table by selecting two tables and the columns containing the shared value. \
-                First, select the columns you want to matching between tables and then select and deselect \
-                columns you want and don't want in your new table. You can only merge tables by joining by \
+        desc:   _t("Create a new dataset by selecting two datasets and the columns containing the shared value. \
+                First, select the columns you want to matching between datasets and then select and deselect \
```
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js]: "First, select the columns you want to matching between datasets and then select and deselect" doesn't sound good to me... It's not quite related to this PR, We can do a better job with this text.

"First, select the matching column in both datasets and then..."

Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js]: Agree, so weird. Let me check it.

File: lib/assets/javascripts/cartodb/dashboard/visualizations/name_visualization_dialog.js [hunk 2]
```diff
@@ -8,11 +8,11 @@
   cdb.admin.NameVisualization = cdb.admin.BaseDialog.extend({
 
     _TEXTS: {
-      title:  _t('Save your visualization'),
-      desc:   _t('A visualization is a mix of layers, styles and sql. Your \
-              visualizations are now accessible from your dashboard.'),
-      button: _t('Create visualization'),
-      error:  _t('The visualization name can\'t be blank')
+      title:  _t('Save your map'),
+      desc:   _t('A map is a mix of layers, styles and sql. Your \
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/dashboard/visualizations/name_visualization_dialog.js]: I would say SQL (in capitals).

File: lib/assets/javascripts/cartodb/table/header/create_visualization_dialog.js [hunk 3]
```diff
@@ -7,11 +7,11 @@
   cdb.admin.CreateVizDialog = cdb.admin.BaseDialog.extend({
 
     _TEXTS: {
-      title:  _t('Save your visualization'),
-      desc:   _t('A visualization is a mix of layers, styles and sql. Your \
-              visualizations are now accessible from your dashboard.'),
-      button: _t('Create visualization'),
-      error:  _t('The visualization name can\'t be blank'),
+      title:  _t('Save your map'),
+      desc:   _t('A map is a mix of layers, styles and sql. Your \
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/table/header/create_visualization_dialog.js]: I would write SQL in capitals

Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/header/create_visualization_dialog.js]: Agree, :+1:

File: lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js (hunk 1 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js]: Sounds better?

**Comment by javierarce** [comment]
Changes look fine :+1:

File: lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js [hunk 4]
```diff
@@ -19,29 +19,29 @@ cdb.admin.MergeTablesModel = cdb.core.Model.extend({
...
     back:       _t("Go back"),
     merge: {
       column: {
         title:  _t("Column Join"),
-        desc:   _t("Create a new table by selecting two tables and the columns containing the shared value. \
-                First, select the columns you want to matching between tables and then select and deselect \
-                columns you want and don't want in your new table. You can only merge tables by joining by \
+        desc:   _t("Create a new dataset by selecting two datasets and the columns containing the shared value. \
+                First, select the matching column in both datasets and then select or deselect \
+                columns you want or not in your new dataset. You can only merge datasets by joining by \
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js]: here

Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js]: Better.

**Comment by alonsogarciapablo** [comment]
I didn't spot anything else. :+1:

File: lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js (hunk 4 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js]: I would change this:

"First, select the matching column in both datasets and then select or deselect columns you want or not in your new dataset." 

with 

"First, select the matching column in both datasets and then select or deselect *the* columns you want in your new dataset." 

So: I'd get rid of the "or not" and I'd add the "the" :)

Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/header/merge_tables_dialog.js]: Lovely :es:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2761/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2762/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2766/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2767/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2768/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2770/))

**Comment by iriberri** [comment]
:+1:

**Pull request merged**